src/Symfony/Component/Form/Tests/ChoiceList/Factory/CachingFactoryDecoratorTest.php

```diff
@@ -25,68 +23,39 @@
  */
 class CachingFactoryDecoratorTest extends TestCase
 {
-    /**
-     * @var MockObject&ChoiceListFactoryInterface
-     */
-    private $decoratedFactory;
-
     /**
      * @var CachingFactoryDecorator
      */
     private $factory;
 
     protected function setUp(): void
     {
-        $this->decoratedFactory = $this->createMock(ChoiceListFactoryInterface::class);
-        $this->factory = new CachingFactoryDecorator($this->decoratedFactory);
+        $this->factory = new CachingFactoryDecorator(new DefaultChoiceListFactory());
     }
 
     public function testCreateFromChoicesEmpty()
     {
-        $list = new ArrayChoiceList([]);
-
-        $this->decoratedFactory->expects($this->once())
-            ->method('createListFromChoices')
-            ->with([])
-            ->willReturn($list);
-
-        $this->assertSame($list, $this->factory->createListFromChoices([]));
-        $this->assertSame($list, $this->factory->createListFromChoices([]));
+        $this->assertEquals(new ArrayChoiceList([]), $this->factory->createListFromChoices([]));
+        $this->assertEquals(new ArrayChoiceList([]), $this->factory->createListFromChoices([]));
```

Comment:
The caching factory should ensure the same instance of list is returned, this test does not assert the same thing anymore and allows regressions.

Reply:
I would revert the changes in this test class.

Reply:
I have reworked the test to always check if the created lists are (not) the same explicitly.

Reply:
Neat!